### PR TITLE
added beta badge

### LINKED ISSUE #3860: [UI] Display badge for beta features
#### Current Behavior
Right now, in meshery ui, there is no way to distinguish beta features from others.

#### Desired Behavior
We need to add a badge for beta features.

#### Implementation
The badge should be added as shown in the mockup for beta features.
For a whole page/component , the badge should be in the navigation bar along with the name.

Beta Components:
1. Meshmap
2. Applications
3. Filters
4. Patterns

#### Mockups
Assuming `Manage service meshes` to be a beta feature, the badge should look like, 

Screenshot 2021-08-12 at 6 40 36 PM

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by ishitakeshawani**
Signed-off-by: ishitakeshawani <keshwaniishita054@gmail.com>

**Description**

This PR fixes #3860

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sudo-NithishKarthik** [comment]
can you add some screenshots @ishitakeshawani  ?

**Review by warunicorn19 — CHANGES_REQUESTED** [review]
Good work @ishitakeshawani, but right now the logo looks quite dull.
Screenshot 2021-08-31 at 4 50 35 PM

Could you please make it transparent, like this 
Screenshot 2021-08-31 at 4 52 10 PM
and then just make the text blue or purple

File: ui/components/Navigator.js [hunk 1]
```diff
@@ -315,6 +315,7 @@ const categories = [
     title : "Configuration",
     show : false,
     link : true,
+    isBeta : true,
```
Comment (warunicorn19) [inline ui/components/Navigator.js:318]: This can be removed, as there's no separate page for Configuration

**Review by warunicorn19 — CHANGES_REQUESTED** [review]

File: ui/components/Navigator.js (hunk 1 above)
Comment (ishitakeshawani) [reply ui/components/Navigator.js:318]: Yes, will remove it.

**Comment by ishitakeshawani** [comment]
> can you add some screenshots @ishitakeshawani ?

Screenshot from 2021-08-31 16-33-54

**Comment by ishitakeshawani** [comment]
> Good work @ishitakeshawani, but right now the logo looks quite dull.
> Screenshot 2021-08-31 at 4 50 35 PM
> 
> Could you please make it transparent, like this
> Screenshot 2021-08-31 at 4 52 10 PM
> and then just make the text blue or purple
> 
> // @sudo-NithishKarthik

Yes, will make it transparent. Should I also put the border color like purple or blue?

**Comment by warunicorn19** [comment]
Yes that would be great

**Comment by ishitakeshawani** [comment]
Is it looking good ?
Screenshot from 2021-08-31 17-27-11

**Comment by warunicorn19** [comment]
Could please make it a little thinner and it would be great if the word "beta" is right in the centre.

**Comment by ishitakeshawani** [comment]
Screenshot from 2021-08-31 18-12-09

**Comment by leecalcote** [comment]
Exciting, @ishitakeshawani. @Nikhil-Ladha's spot on with respect to the color.

**Comment by ishitakeshawani** [comment]
Which color should I use here? There is also some whitespace below the text beta. Inspected it many times but have no idea that how to remove it.
Screenshot from 2021-09-01 16-45-02

**Comment by ishitakeshawani** [comment]
Screenshot from 2021-09-01 16-49-49

**Comment by ishitakeshawani** [comment]
@warunicorn19 @sudo-NithishKarthik

**Comment by sudo-NithishKarthik** [comment]
Yes, i think that too will work...

**Comment by ishitakeshawani** [comment]
> @ishitakeshawani I was thinking maybe we could use something like this instead?
> Screenshot_20210902-002549
> 
> @sudo-NithishKarthik @Nikhil-Ladha @debo19 views?
> This is from youtube btw

Do you mean by the logo, then meshery and beta all?

**Comment by warunicorn19** [comment]
No i mean the `beta` would go one top of `Applications` like in the image.

**Comment by ishitakeshawani** [comment]
> No I mean the `beta` would go one top of `Applications` like in the image.

Ok got it. Sounds good.

**Comment by leecalcote** [comment]
There are orange / yellow colors included in the Meshery UI Figma file that can be used for the "beta" word.

**Comment by ishitakeshawani** [comment]
> There are orange/yellow colors included in the Meshery UI Figma file that can be used for the "beta" word.

Yes, I will use these colors. Thank you @leecalcote

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
Thank you for this @ishitakeshawani , but this is not a scalable approach, we are planning to use it for components as well as for pages, so, we have to implement it in such a way that it does not depend on whether we want to apply this to a page or to a component.

File: ui/lib/store.js [hunk 2]
```diff
@@ -176,9 +178,9 @@ export const updatepagepath = ({path}) => dispatch => {
   return dispatch({ type: actionTypes.UPDATE_PAGE, path });
 }
 
-export const updatepagetitle = ({path, title}) => dispatch => {
+export const updatepagetitle = ({path, title, isBeta}) => dispatch => {
```
[the remark this answers is not in the document]
Comment (ishitakeshawani) [reply ui/lib/store.js:181]: Ok, I will use another approach as you mention. Thank you for pointing it.

**Comment by leecalcote** [comment]
@ishitakeshawani heads-up - a merge conflict to address.

**Comment by ishitakeshawani** [comment]
> @ishitakeshawani heads-up - a merge conflict to address.

Oh, will solve it by tonight.  Thank you, Lee.

**Comment by leecalcote** [comment]
Thank you, @ishitakeshawani. You don't happen to have a screenshot handy, do you?

**Comment by ishitakeshawani** [comment]
> Thank you, @ishitakeshawani. You don't happen to have a screenshot handy, do you?

Sorry, for the delay. I will send the screenshot and push the code by tomorrow.

**Comment by ishitakeshawani** [comment]
@sudo-NithishKarthik  I did changes in the code. Can you review it. In the header, we need to check if the page is beta then add the badge. But in the component, it is possible to add a direct beta badge in code. So should I do it or what approach should  I use for components ?

Screenshot from 2021-09-19 16-46-47
Screenshot from 2021-09-19 16-46-23

**Comment by leecalcote** [comment]
I updated the color. Visually, the tag looks good to me. @sudo-NithishKarthik, are we good to merge with this beta tagging framework?

**Review by leecalcote — DISMISSED** [review]

**Comment by sudo-NithishKarthik** [comment]
Sorry for the late response. @ishitakeshawani , although we can improve it further, let's not spend too much time on this. This will work just fine. Thank you.

**Review by sudo-NithishKarthik — APPROVED** [review]

**Comment by ishitakeshawani** [comment]
> Sorry for the late response. @ishitakeshawani , although we can improve it further, let's not spend too much time on this. This will work just fine. Thank you.

Thank you @sudo-NithishKarthik Looking forward to learn more.

**Pull request merged**